crates/cheatcodes/src/test/revert_handlers.rs

```diff
@@ -108,7 +108,16 @@ fn handle_revert(
         } else {
             (&stringify(&actual_revert), &stringify(expected_reason))
         };
-        Err(fmt_err!("Error != expected error: {} != {}", actual, expected,))
+
+        // When using vm.expectRevert(abi.encodeWithSignature("Error(string)", "A"));, the expected
+        // reason starts with "revert: " We strip redundant `revert: ` prefix from the revert reason
+        let expected = &expected.replace("revert: ", "");
```

Comment:
I had to do the check here because I need the decoded "actual" value.

It relies on the `&decoder.decode` which has: `Note that this is just a best-effort guess, and should not be relied upon for anything other than user output.`

I can add a specific check for this `0x08c379a0`: the function selector for Error(string) (i.e., keccak256("Error(string)")[:4] above but i think it adds overload but lmk what you think.

Reply:
To avoid relying purely on best-effort decoding, good to add a check for the `0x08c379a0` selector (Error(string)), which is the most common revert pattern. This keeps decoding accurate without affecting performance much.